docs/overview/.nav.yml

```diff
@@ -0,0 +1,12 @@
+# `.nav.yml` is provided by https://lukasgeiter.github.io/mkdocs-awesome-nav
+nav:
+  - Welcome to Django Components: welcome.md
```

Comment:
I've moved the titles into the new nav files. Doesn't matter much, IMO, but at least the title in the navigation will match the title at the top of the page.

E.g. in this screenshot below, the main title is using `code blocks`, whereas the title on the side does not.

<img width="776" alt="Screenshot 2025-02-23 at 22 39 23" src="https://github.com/user-attachments/assets/0b9bb384-f0be-4cd2-a734-a8d65805a47c" />